invenio_pidstore/providers/datacite.py

```diff
@@ -31,20 +35,66 @@ class DataCiteProvider(BaseProvider):
     default_status = PIDStatus.NEW
     """Default status for newly created PIDs by this provider."""
 
+    doi_prefix_regexp = re.compile(
+        r"10\.\d+(\.\d+)*$"
+    )
+
     @classmethod
-    def create(cls, pid_value, **kwargs):
+    def valid_doi_prefix(cls, prefix):
+        """Matches if prefix is a DOI prefix.
+
+        NOTE: This was done here to prevent relying on idutils.
+        """
+        return cls.doi_prefix_regexp.match(prefix)
+
+    @classmethod
+    def generate_doi(cls, options=None):
+        """Generate DOI with random suffix."""
+        passed_options = options or {}
+        # WHY: A new dict needs to be created to prevent side-effects
+        options = dict(current_app.config.get(
```

Comment:
Watch out here because you are not doing a `deepcopy` but instead it is a shallow copy, so if you have mutable object inside they are not copied, but only referenced (see [here](https://www.peterbe.com/plog/be-careful-with-using-dict-to-create-a-copy) first link I found).
Maybe this is what you want, but the `WHY` comment is a bit misleading IMHO

Reply:
Yes, thanks for that! Switched to `copy.deepcopy`.